lib/pbench/test/unit/server/query_apis/test_datasets_contents.py

```diff
@@ -472,3 +472,39 @@ def test_get_index(self, attach_dataset, provide_metadata):
         drb = Dataset.query(name="drb")
         indices = self.cls_obj.get_index(drb, self.index_from_metadata)
         assert indices == "unit-test.v6.run-toc.2020-05"
+
+    def test_with_invalid_run_id(self, client, server_config, pbench_token):
+        """
+        Check the Contents API when invalid run_id is provided
+        """
+
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.NOT_FOUND
+
+    def test_with_empty_run_id(self, client, server_config, pbench_token):
+        """
+        Check the Contents API when no run_id is provided
+        """
+        self.payload["run_id"] = ""
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.NOT_FOUND
+
+    def test_without_run_id(self, client, server_config, pbench_token):
+        """
+        Check the Contents API when no run_id is provided
+        """
+        del self.payload["run_id"]
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
```

Comment:
That's not a bad idea, @npalaska, but you code needs a small tweak, because both `""` and `None` are false-y, and we only want to delete the mapping if `run_id` is actually `None`.

Reply:
You are right, should be:
```
    if run_id is None:
        del self.payload["run_id"]
    else:
        self.payload["run_id"] = run_id
```